cpp/modmesh/buffer/SimpleArray.cpp

```diff
@@ -28,56 +28,46 @@
 
 #include <modmesh/buffer/SimpleArray.hpp>
 
+#include <unordered_map>
+
 namespace modmesh
 {
 
-DataType get_data_type_from_string(const std::string & data_type_string)
+namespace detail
 {
-    if (data_type_string == "bool")
-    {
-        return DataType::Bool;
-    }
-    if (data_type_string == "int8")
-    {
-        return DataType::Int8;
-    }
-    if (data_type_string == "int16")
-    {
-        return DataType::Int16;
-    }
-    if (data_type_string == "int32")
-    {
-        return DataType::Int32;
-    }
-    if (data_type_string == "int64")
-    {
-        return DataType::Uint64;
-    }
-    if (data_type_string == "uint8")
-    {
-        return DataType::Uint8;
-    }
-    if (data_type_string == "uint16")
-    {
-        return DataType::Uint16;
-    }
-    if (data_type_string == "uint32")
-    {
-        return DataType::Uint32;
-    }
-    if (data_type_string == "uint64")
-    {
-        return DataType::Uint64;
-    }
-    if (data_type_string == "float32")
-    {
-        return DataType::Float32;
-    }
-    if (data_type_string == "float64")
+struct DataTypeHasher
+{
+    std::size_t operator()(const std::string & data_type_string) const
+    {
+        const char ch1 = data_type_string.front();
+        const char ch2 = data_type_string.back();
+        return static_cast<std::size_t>(ch1) << 8 | static_cast<std::size_t>(ch2);
+    }
+};
```

Comment:
End marking.  Please don't forget about it.  It's tedious to pay attention to basic styles like this.